### PR TITLE
added backup function to database object

### PR DESCRIPTION
**Pull request by schraf**
I wanted to have the backup support from sqlite3 exposed in node-sqlite3 for a project I am working on. Here is a link to the documentation for this API: [Using the SQLite Online Backup API](https://sqlite.org/backup.html). I have included a unit test that just makes a backup of an existing database in the test/support directory. I only tested this on my ArchLinux server.

### CODE REVIEW

**Comment by laysdra7265** [comment]
I got an `Error: SQLITE_CANTOPEN: library routine called out of sequence` when trying to use this PR commit to backup sqlite database file.
node version: v6.11.3
system version: centos 7

**Comment by laysdra7265** [comment]
Actually the problem is the path not existed,we need to create the path first.
Or get a `library routine called out of sequence` error without any useful error message.

File: src/database.cc [hunk 1]
```diff
@@ -371,6 +374,83 @@ NAN_METHOD(Database::Interrupt) {
...
+void Database::Work_AfterBackup(uv_work_t* req) {
+    Nan::HandleScope scope;
+
+    BackupBaton* baton = static_cast<BackupBaton*>(req->data);
+    Database* db = baton->db;
+
+    Local<Function> cb = Nan::New(baton->callback);
+
+    if (baton->status != SQLITE_OK) {
+        EXCEPTION(Nan::New(baton->message.c_str()).ToLocalChecked(), baton->status, exception);
```
Comment (kewde) [inline src/database.cc]: > EXCEPTION(baton->message, baton->status, exception);

File: test/backup.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+var assert = require('assert');
+
+describe('backup', function() {
+    var db;
+    before(function(done) {
+        db = new sqlite3.Database('test/support/prepare.db', sqlite3.OPEN_READONLY, done);
+    });
+
+    it('backup database', function(done) {
+        db.backup('test/support/backup.db', done);
```
Comment (kewde) [inline test/backup.test.js]: Seems a bit of a thin test case.
I don't mind improving it if anyone has ideas of what could get wrong.

File: src/database.cc (hunk 1 above)
Comment (kewde) [reply src/database.cc]: To make it comply with V8 7.1

**Comment by paulfitz** [comment]
@schraf thanks for this, it was helpful. I extended it in #1116 to support the backup api more completely.

**Pull request closed without merging**